registrations/serializers.py

```diff
@@ -274,14 +273,131 @@ def validate(self, data):
 
 class SeatReservationCodeSerializer(serializers.ModelSerializer):
     timestamp = DateTimeField(default_timezone=pytz.UTC, required=False)
+
     expiration = serializers.SerializerMethodField()
 
-    class Meta:
-        fields = ("seats", "code", "timestamp", "registration", "expiration")
-        model = SeatReservationCode
+    in_waitlist = serializers.SerializerMethodField()
 
     def get_expiration(self, obj):
-        return obj.timestamp + timedelta(minutes=code_validity_duration(obj.seats))
+        return obj.expiration
+
+    def get_in_waitlist(self, obj):
+        registration = obj.registration
+        maximum_attendee_capacity = registration.maximum_attendee_capacity
+
+        if maximum_attendee_capacity is not None:
+            attendee_count = registration.signups.filter(
+                attendee_status=SignUp.AttendeeStatus.ATTENDING
+            ).count()
+
+            if maximum_attendee_capacity - attendee_count <= 0:
+                return True
+
+        return False
+
+    def validate(self, data):
+        instance = self.instance
+        errors = {}
+
+        # Amount of seats is always required
+        if data.get("seats") is None:
+            errors["seats"] = ErrorDetail(
+                _("This field must be specified."), code="required"
+            )
+
+        if isinstance(instance, SeatReservationCode):
+            non_editable_fields = ["registration", "timestamp"]
```

Comment:
You can define read-only fields in the serializer meta https://www.django-rest-framework.org/api-guide/serializers/#specifying-read-only-fields

Reply:
Ah, but I guess you need `registration` here to be able to create an seat reservation. :thinking: One option probably would be to have `validate_registration` or define `registration` dynamically as readonly in `__init__` :thinking:

Reply:
I ended up to have timestamp as read_only field and preventing to change registration in to_internal_value method

```
timestamp = DateTimeField(default_timezone=pytz.UTC, required=False, read_only=True)
```

```
def to_internal_value(self, data):
        if self.instance:
            data["registration"] = self.instance.registration.id
        return super().to_internal_value(data)
```